### PR TITLE
Update kernelx64.config to support UFS

### PR DESCRIPTION
**Pull request by darksidemilk**
UFS support for microsoft surface go 4 in x64 kernel
Enabled UFSHCD_PCI driver to support universal flash storage on PCI devices
See also https://forums.fogproject.org/topic/17112/surface-go-4-incompatible/3?_=1699986746721

### CODE REVIEW

File: configs/kernelx64.config [hunk 1]
```diff
@@ -3085,7 +3085,7 @@ CONFIG_MMC_MTK=y
 CONFIG_SCSI_UFSHCD=y
 # CONFIG_SCSI_UFS_BSG is not set
```
Comment (rluzuriaga) [inline configs/kernelx64.config:3086]: @darksidemilk Have you checked if adding `CONFIG_SCSI_UFS_BSG` gives you any positive, or negative, results?
From the little I've researched, this config may need to be enabled for us to use [`ufs-tool`](https://github.com/westerndigitalcorporation/ufs-utils), which is probably the tool we'll need to use to fully support UFS.

**Comment by darksidemilk** [comment]
My first test I enabled that one, the PCI one in the pull as well as the top level "subsystem" option. It worked, I just then tested down to just PCI to get the minimal requirement to have the disk recognized and imaged in fog.
I have no objections to including the bcg and or subsystem options. Our past documentation mentions limiting modules added to the kernel to keep maintenance simple, so I tried to follow that.



Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Rodrigo Luzuriaga ***@***.***>
Sent: Saturday, November 18, 2023 6:37:18 PM
To: FOGProject/fos ***@***.***>
Cc: JJ Fullmer ***@***.***>; Mention ***@***.***>
Subject: [External]Re: [FOGProject/fos] Update kernelx64.config to support UFS (PR #78)


@rluzuriaga commented on this pull request.

________________________________

In configs/kernelx64.config<https://github.com/FOGProject/fos/pull/78#discussion_r1398304978>:

> @@ -3085,7 +3085,7 @@ CONFIG_MMC_MTK=y
 CONFIG_SCSI_UFSHCD=y
 # CONFIG_SCSI_UFS_BSG is not set


@darksidemilk<https://github.com/darksidemilk> Have you checked if adding CONFIG_SCSI_UFS_BSG gives you any positive, or negative, results?
From the little I've researched, this config may need to be enabled for us to use ufs-tool<https://github.com/westerndigitalcorporation/ufs-utils>, which is probably the tool we'll need to use to fully support UFS.

**Comment by darksidemilk** [comment]
I also intend to update the compile fos kernel doc with whatever we add to the config

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Rodrigo Luzuriaga ***@***.***>
Sent: Saturday, November 18, 2023 6:37:18 PM
To: FOGProject/fos ***@***.***>
Cc: JJ Fullmer ***@***.***>; Mention ***@***.***>
Subject: [External]Re: [FOGProject/fos] Update kernelx64.config to support UFS (PR #78)


@rluzuriaga commented on this pull request.

________________________________

In configs/kernelx64.config<https://github.com/FOGProject/fos/pull/78#discussion_r1398304978>:

> @@ -3085,7 +3085,7 @@ CONFIG_MMC_MTK=y
 CONFIG_SCSI_UFSHCD=y
 # CONFIG_SCSI_UFS_BSG is not set


@darksidemilk<https://github.com/darksidemilk> Have you checked if adding CONFIG_SCSI_UFS_BSG gives you any positive, or negative, results?
From the little I've researched, this config may need to be enabled for us to use ufs-tool<https://github.com/westerndigitalcorporation/ufs-utils>, which is probably the tool we'll need to use to fully support UFS.

**Comment by rluzuriaga** [comment]
@darksidemilk I just wanted to check if anything broke from your testing with that option enabled, especially because I don't have a machine with UFS to test. I agree that we should keep the enabled options to the minimum required, so I don't plan on enabling bsg unless we need it in the future.

**Comment by darksidemilk** [comment]
Nothing broke with that option enabled on the ufs device. The drive was recognized with and without it.
I haven't yet tested it on other devices though. I plan to first thing tomorrow



Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________
From: Rodrigo Luzuriaga ***@***.***>
Sent: Sunday, November 19, 2023 4:56:41 PM
To: FOGProject/fos ***@***.***>
Cc: JJ Fullmer ***@***.***>; Mention ***@***.***>
Subject: [External]Re: [FOGProject/fos] Update kernelx64.config to support UFS (PR #78)


@darksidemilk<https://github.com/darksidemilk> I just wanted to check if anything broke from your testing with that option enabled, especially because I don't have a machine with UFS to test. I agree that we should keep the enabled options to the minimum required, so I don't plan on enabling bsg unless we need it in the future.

**Comment by darksidemilk** [comment]
I have confirmed that there are not problems with Non-UFS hardware with this added to the kernel as far as I have found.
However, the surface go4 ufs drive is defaulting to a 4k block size, so images captured from say a vmware VM that has a 512e block size are note booting once deployed.

**Pull request merged**